From e30076690b82e000077b664275234d0a36a373e4 Mon Sep 17 00:00:00 2001
From: Mandy Lavi <mandy.lavi@freescale.com>
Date: Thu, 25 Jun 2015 17:17:11 +0300
Subject: [PATCH 247/451] fmd: workaround: force cache coherency on every hc
 FD tx

Signed-off-by: Mandy Lavi <mandy.lavi@freescale.com>
[Xulin: Original patch taken from
Linux-LS1043A-SDK-V0.4-SOURCE-20150826-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../freescale/fman/src/wrapper/lnxwrp_fm_port.c    |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/fman/src/wrapper/lnxwrp_fm_port.c b/drivers/net/ethernet/freescale/fman/src/wrapper/lnxwrp_fm_port.c
index b7a8b5a..7dfe2c7 100644
--- a/drivers/net/ethernet/freescale/fman/src/wrapper/lnxwrp_fm_port.c
+++ b/drivers/net/ethernet/freescale/fman/src/wrapper/lnxwrp_fm_port.c
@@ -217,6 +217,10 @@ static void FqFree(struct qman_fq *fq)
 	XX_Free((t_FmTestFq *) fq);
 }
 
+#warning this is a temporary workaround
+/* should eventually be removed after coherency issue is fixed */
+#include "../../../../../../arch/arm64/include/asm/cacheflush.h"
+
 static t_Error QmEnqueueCB(t_Handle h_Arg, void *p_Fd)
 {
 	t_LnxWrpFmDev *p_LnxWrpFmDev = (t_LnxWrpFmDev *) h_Arg;
@@ -254,8 +258,7 @@ static t_Error QmEnqueueCB(t_Handle h_Arg, void *p_Fd)
 }
 #endif
 
-
-printk("QmEnqueueCB>  p_Fd->addr 0x%llx \n",((struct qm_fd *) p_Fd)->addr);
+	flush_cache_all();
 
 	_errno = qman_enqueue(p_LnxWrpFmDev->hc_tx_fq, (struct qm_fd *) p_Fd,
 			      0);
-- 
1.7.5.4

